packages/workspace/src/common/workspace-protocol.ts

```diff
@@ -32,4 +32,9 @@ export interface WorkspaceServer {
      */
     setRoot(uri: string): Promise<void>;
 
+    /**
+     * Returns list of recents opened workspace as a string []
+     */
+    getRecentRoots(): Promise<string[]>
```

Comment:
I think, we should rename methods to `getWorkspace/setWorkspace/getRecentWorkspaces`. They are not roots. See also: https://github.com/theia-ide/theia/pull/2117#discussion_r199396582